Add 2.3.2 release notes (#6719)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## 2.3.2

### Enhancements

> * UI displays a message when backend is unreachable - https://github.com/PrefectHQ/prefect/pull/6670

Reviewer: ```suggestion
* UI displays an error message when backend is unreachable - https://github.com/PrefectHQ/prefect/pull/6670
```

### Fixes
* Fix issue where parameters weren't updated when a deployment was re-applied by @lennertvandevelde in https://github.com/PrefectHQ/prefect/pull/6668
* Fix issues with stopping Orion on Windows machines - https://github.com/PrefectHQ/prefect/pull/6672
* Fix issue with GitHub storage running in non-empty directories - https://github.com/PrefectHQ/prefect/pull/6693

> * New deployments now respect all user inputs - https://github.com/PrefectHQ/prefect/pull/6695

Author: How about: 
"Fix issue where some user-supplied values were overridden when creating new deployments"?

### Collections

> * [prefect-fugue](https://prefecthq.github.io/prefect-jupyter/)

Reviewer: ```suggestion
* Added [prefect-fugue](https://prefecthq.github.io/prefect-jupyter/) 
```

### Contributors

> * @lennertvandevelde made their first contribution — [https://github.com/PrefectHQ/prefect/pull/6668](https://github.com/PrefectHQ/prefect/pull/6668)

Reviewer: ```suggestion
* @lennertvandevelde made their first contribution! — [https://github.com/PrefectHQ/prefect/pull/6668](https://github.com/PrefectHQ/prefect/pull/6668)
```
